### PR TITLE
DOP-3691

### PR DESCRIPTION
**Pull request by seungpark**
**Goals:**
- update mut-index to transfer `document.facets<json>` from output of parser to `manifest.documents[].facets<json>`
- the parser takes care of nested objects, and mut should read and copy. the conversion to Atlas Search Index keys should happen in the search-transport-server, as this is central to any `index key<->nested` json conversion (the index generation and query parsing happens here)
- included bson in test is generated from a [test doc](https://github.com/seungpark/test-repo/blob/DOP-3691/source/index.txt) using the [parser branch for DOP-3688](https://github.com/mongodb/snooty-parser/tree/DOP-3688-validation)

### CODE REVIEW

File: mut/index/SnootyManifest.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ class ManifestEntry(TypedDict):
     code: dict
     preview: Optional[str]
     tags: List[str]
+    facets: Optional[Dict[str, any]]
```
Comment (i80and) [inline mut/index/SnootyManifest.py:23]: This needs to be `Any`, and must also be imported from typing on line 10

File: mut/index/SnootyManifest.py [hunk 2]
```diff
@@ -121,6 +123,14 @@ def derive_preview(self) -> Optional[str]:
         # Give up and just don't provide a preview.
         else:
             return None
+    
+    def derive_facets(self) -> Optional[Dict[str, any]]:
```
Comment (i80and) [inline mut/index/SnootyManifest.py:127]: ditto

**Review by i80and — CHANGES_REQUESTED** [review]
Can you run `poetry run black mut/index/` to format this file after making the typing changes? Unfortunately, automated linting isn't set up in our github action at this point :(

File: mut/index/SnootyManifest.py (hunk 1 above)
Comment (seungpark) [reply mut/index/SnootyManifest.py]: great catch! updated

File: mut/index/SnootyManifest.py (hunk 2 above)
Comment (seungpark) [reply mut/index/SnootyManifest.py]: same :)

**Comment by seungpark** [comment]
> Can you run `poetry run black mut/index/` to format this file after making the typing changes? Unfortunately, automated linting isn't set up in our github action at this point :(

done! surprised we didn't have a CI error for it 😢

File: mut/index/SnootyManifest.py [hunk 3]
```diff
@@ -142,7 +142,9 @@ def find_metadata(self):
         results = jsonpath_expr.find(self.tree)
         if results:
             results = results[0].value
-            if "robots" in results and (results["robots"] == "None" or "noindex" in results["robots"]):
+            if "robots" in results and (
+                results["robots"] == "None" or "noindex" in results["robots"]
```
Comment (i80and) [inline mut/index/SnootyManifest.py:146]: does case matter for the `"None"` comparison?

**Review by i80and — APPROVED** [review]

File: mut/index/SnootyManifest.py (hunk 3 above)
Comment (seungpark) [reply mut/index/SnootyManifest.py:146]: this is change from black formatting and should not change logic here... that said, i dont think case matters here. None, as in from parser output, i believe

**Pull request merged**